src/scrapers/hapoalim.js

```diff
@@ -55,25 +55,25 @@ async function fetchAccountData(page, options) {
     let summary;
     try {
       const balanceCreditResult = await fetchGetWithinPage(page, balanceCreditUrl);
-      summary = fetchSummary(balanceCreditResult);
+      summary = convertToSummary(balanceCreditResult);
+
+      const txnsUrl = `${apiSiteUrl}/current-account/transactions?accountId=${accountNumber}&numItemsPerPage=150&retrievalEndDate=${endDateStr}&retrievalStartDate=${startDateStr}&sortCode=1`;
+      let txns;
+      try {
+        const txnsResult = await fetchGetWithinPage(page, txnsUrl);
+        txns = convertTransactions(txnsResult.transactions);
+      } catch (err) {
+        txns = [];
+      }
```

Comment:
@chilikla same here for `try..catch` - we don't need it as we're covered by the base scrapers in case something fails.

Reply:
@eshaham 
Sorry for the absence lately, but I returned :relaxed:

Regarding to getting rid of the two `try... catch` blocks, I understand your point, but there are 2 scenarios which it will be problematic in that case:

- In case of moving accounts between branches, Hapoalim is keeping the old accounts accessible through their website (along with the new active ones), accessible but empty. Their particular API pages are empty and therefore failing the summary fetch action, even though there is no actual problem since the active accounts already fetched. Using `try... catch` mechanism will return the real accounts data and won't fail the whole scarping.

- Accounts with not transactions will fail the scraper as well, in the transactions fetch actions. And I'm not sure it won't be better to just return empty array of transactions, since there is no problem with the account per se, it just silent.

What do you think?

Reply:
> Their particular API pages are empty and therefore failing the summary fetch action, even though there is no actual problem since the active accounts already fetched.

Got it. Then while I usually avoid comments in code, this is a classic place to add a comment explaining why the `try...catch` is needed here.

> Accounts with not transactions will fail the scraper as well, in the transactions fetch actions

Assuming this is indeed the case for accounts without transactions, then same response as above 😄
Once you add a comment I think we can merge this.